e2e: TestKuke_AttachDetach_KeepsTaskRunning times out in waitForSocket on origin/main

## Repro

Reproduced on `origin/main` at commit `f4447f8` via clean worktree:

```bash
git worktree add /tmp/kukeon-main-check origin/main
cd /tmp/kukeon-main-check
go build -o kuke ./cmd/
sudo env PATH=/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    HOME=$HOME E2E_BIN_DIR=$(pwd) \
    /usr/local/go/bin/go test -v -count=1 \
    -run TestKuke_AttachDetach_KeepsTaskRunning ./e2e
```

Host has `sbsh` on PATH (passes the existing `requireSbshHasPostMergeFlags` guard — `sbsh terminal --help` lists `--capture-file`).

## Observed

```
=== RUN   TestKuke_AttachDetach_KeepsTaskRunning
=== PAUSE TestKuke_AttachDetach_KeepsTaskRunning
=== CONT  TestKuke_AttachDetach_KeepsTaskRunning
    e2e_kuke_attach_test.go:173: sbsh socket "/tmp/kukeon-main-check/e2e/tmp/.../work/tty/socket" did not appear within 10s
--- FAIL: TestKuke_AttachDetach_KeepsTaskRunning (22.91s)
FAIL    github.com/eminwux/kukeon/e2e   22.930s
```

Failure lands on `waitForSocket` in `e2e/e2e_kuke_attach_test.go:171-173` — i.e. **before** `kuke attach` is even invoked. After `applyAttachableCell` returns Ready, the per-container sbsh control socket at `<runPath>/<realm>/<space>/<stack>/<cell>/work/tty/socket` never appears within the 10s grace.

## Root cause

Not yet pinpointed — this issue is filed to track the investigation. The symptom narrows the surface to:

- The in-container `work` task is started by `applyAttachableCell` (which exits successfully and reports the cell Ready), so the OCI bundle and runner contract appear fine, but
- The expected `tty/socket` inside the per-container `tty/` bind-mount target never gets bound by sbsh inside the container within 10s.

Candidates for where the bug lives:

1. The sbsh-terminal wrapper command kukeon injects as the container's argv (`internal/ctr`, post-#69 + sbsh#153). If a flag was added/renamed in a newer sbsh, `sbsh terminal` may be exiting before the bind, and the existing `requireSbshHasPostMergeFlags` guard only checks for `--capture-file`.
2. The per-container `tty/` bind-mount (#77) — the source/target mode or ownership may be regressing such that sbsh's `bind(2)` fails silently inside the container.
3. The `attachable-cell.yaml` fixture pulls `registry.eminwux.com/busybox:latest`; if the image is unpacked but lacks the entry sbsh expects, the wrapper could no-op.

## Suggested fix

Reproduce, then capture the in-container task's stderr (the runner currently swallows it once the bind fails) by redirecting via the wrapper or running `ctr -n <realm> task ls/exec` while the test sleeps. Once the actual sbsh failure mode is visible, fix the root cause and tighten `requireSbshHasPostMergeFlags` (or its equivalent) to assert the specific flag/version invariant the wrapper depends on, so a stale sbsh on PATH skips loudly instead of timing out at `waitForSocket`.

## Notes

Surfaced while validating PR #115 (kuke attach via pkg/attach). The PR's diff is purely client-side (`cmd/kuke/attach`), so the failure is unrelated to that change — both `origin/main` and the PR branch fail identically at the same step. The PR body defers e2e re-run to this fix.

--- a/e2e/e2e_kuke_attach_test.go
+++ b/e2e/e2e_kuke_attach_test.go
@@ -90,21 +90,25 @@ func stageHostSbsh(t *testing.T, runPath string) {
 	}
 }
 
-// requireSbshHasPostMergeFlags ensures the host sbsh carries the flag set
-// the wrapper injected at #69 assumes (post sbsh#153, which added
-// `--capture-file` to the `sbsh terminal` subcommand). Without this guard a
-// stale sbsh on PATH manifests as `unknown flag: ...` deep inside the work
-// container's task, surfaced here only as a `waitForSocket` timeout — hard
-// to attribute. Skipping with a named cause saves the dig.
+// requireSbshHasPostMergeFlags ensures the host sbsh carries every flag the
+// in-container wrapper (`internal/ctr/attachable.go`) injects into the
+// workload's argv. Without this guard a stale or newer-renamed sbsh on PATH
+// manifests as `unknown flag: ...` deep inside the work container's task,
+// surfaced here only as a `waitForSocket` timeout — hard to attribute.
+// Skipping with a named cause saves the dig.
 func requireSbshHasPostMergeFlags(t *testing.T, hostSbsh string) {
 	t.Helper()
 	out, err := exec.Command(hostSbsh, "terminal", "--help").CombinedOutput()
 	if err != nil {
 		t.Fatalf("sbsh terminal --help failed: %v\noutput:\n%s", err, out)
 	}
-	if !strings.Contains(string(out), "--capture-file") {
-		t.Skipf("host sbsh %q lacks `sbsh terminal --capture-file` (sbsh#153); "+
-			"upgrade sbsh on PATH and retry", hostSbsh)
+	help := string(out)
+	for _, flag := range []string{"--socket", "--capture-file", "--log-file"} {
+		if !strings.Contains(help, flag) {
+			t.Skipf("host sbsh %q lacks `sbsh terminal %s`; "+
+				"upgrade or downgrade sbsh on PATH to a build whose `terminal` "+
+				"subcommand carries every wrapper flag and retry", hostSbsh, flag)
+		}
 	}
 }
 

--- a/internal/ctr/attachable.go
+++ b/internal/ctr/attachable.go
@@ -122,9 +122,9 @@ func withAttachableArgsWrap() oci.SpecOpts {
 			AttachableBinaryPath,
 			AttachableSubcommand,
 			"--run-path", AttachableTTYDir,
-			"--terminal-socket", AttachableSocketPath,
+			"--socket", AttachableSocketPath,
 			"--capture-file", AttachableCapturePath,
-			"--terminal-logfile", AttachableLogfilePath,
+			"--log-file", AttachableLogfilePath,
 			"--",
 		}
 		s.Process.Args = append(wrapped, original...)

--- a/internal/ctr/attachable_test.go
+++ b/internal/ctr/attachable_test.go
@@ -192,9 +192,9 @@ func TestBuildContainerSpec_AttachableTrue_MountsAndArgsWrap(t *testing.T) {
 				ctr.AttachableBinaryPath,
 				ctr.AttachableSubcommand,
 				"--run-path", ctr.AttachableTTYDir,
-				"--terminal-socket", ctr.AttachableSocketPath,
+				"--socket", ctr.AttachableSocketPath,
 				"--capture-file", ctr.AttachableCapturePath,
-				"--terminal-logfile", ctr.AttachableLogfilePath,
+				"--log-file", ctr.AttachableLogfilePath,
 				"--",
 			}
 			if len(spec.Process.Args) < len(wantPrefix) {
@@ -235,9 +235,9 @@ func TestBuildContainerSpec_AttachableTrue_EmptyImageArgs(t *testing.T) {
 		ctr.AttachableBinaryPath,
 		ctr.AttachableSubcommand,
 		"--run-path", ctr.AttachableTTYDir,
-		"--terminal-socket", ctr.AttachableSocketPath,
+		"--socket", ctr.AttachableSocketPath,
 		"--capture-file", ctr.AttachableCapturePath,
-		"--terminal-logfile", ctr.AttachableLogfilePath,
+		"--log-file", ctr.AttachableLogfilePath,
 		"--",
 	}
 	if !reflect.DeepEqual(spec.Process.Args, want) {

